flycheck.el

```diff
@@ -4214,6 +4214,20 @@ universal prefix arg, and only the id with normal prefix arg."
     (`(eval ,_) t)
     (_ nil)))
 
+(defun flycheck-cwd-wrapper (checker)
+  "Function returning apropriate current working directory.
+
+Work out the working directory from which the CHECKER command
+will be called.  The default is the `default-directory' of the source-buffer
+from which flycheck is calling the checker.
+:cwd can be a list that contains a fixed path string or a function that will
+return a path.  The function needs to return an absolut path not a relative path."
+  (let ((cwd (flycheck-checker-get checker 'cwd)))
+    (or (and (functionp cwd) (funcall cwd))
+        (and (listp cwd) (stringp (car cwd)) (file-truename (car cwd)))
```

Comment:
@s3bs I think that's a very obscure corner case, and I'm not sure whether it's reasonable to add an extra special case just for this purpose, the more since if you really need it you could always write a function that just returns the constant string.

I'd really prefer if we were able to remove this special case from this implementation, and only accept functions for the working directory.  That'd also somewhat simplify this code.